base/applications/regedit/find.c

```diff
@@ -223,16 +231,19 @@ BOOL RegFindRecurse(
                                   NULL, &cb);
         if (lResult != ERROR_SUCCESS)
             goto err;
-        pb = malloc(cb);
+        pb = malloc(cb + 3); /* To avoid buffer overrun, append 3 NULs */
         if (pb == NULL)
             goto err;
         lResult = RegQueryValueExW(hSubKey, ppszNames[i], NULL, &type,
                                   pb, &cb);
         if (lResult != ERROR_SUCCESS)
             goto err;
 
+        /* To avoid buffer overrun, append 3 NULs */
+        pb[cb] = pb[cb + 1] = pb[cb + 2] = 0;
```

Comment:
Suggestion if you follow above comment: pb[cb] = pb[cb + 1] = 0;

Note: It is interesting that I have been working on this exact same area of the code. I have found that there are some values in the registry that are not NULL terminated and this causes crashes. With this approach, it should be fixed.

Reply:
`cb` can be an odd number. 2 bytes is not enough.